### PR TITLE
Move quote token

### PR DESCRIPTION
**Pull request by grandizzy**
- create target bucket if not initialized
- move tokens with up reallocation from lower to upper bucket
- update lup if lup in (from, to) bucket interval

### CODE REVIEW

File: src/libraries/Buckets.sol [hunk 1]
```diff
@@ -293,49 +346,46 @@ library Buckets {
 
     function reallocateUp(
```
Comment (MikeHathaway) [inline src/libraries/Buckets.sol:347]: Does `reallocateDown` need to be modified as well with explicit passing of the `fromBucket` and `toBucket`?

**Review by MikeHathaway — APPROVED** [review]
Question about the reallocate interface, but generally LGTM

File: src/libraries/Buckets.sol (hunk 1 above)
Comment (grandizzy) [reply src/libraries/Buckets.sol:347]: I don't think so, only `reallocateUp` is bounded by a fom and to bucket, `reallocateDown` starts from a bucket and reallocate as long as there is debt to reallocate

File: src/libraries/Buckets.sol [hunk 2]
```diff
@@ -67,6 +68,58 @@ library Buckets {
...
+        uint256 removeFromDeposit = Maths.min(_amount, fromBucket.onDeposit);
+        fromBucket.onDeposit -= removeFromDeposit;
+
+        Bucket storage toBucket = buckets[_toPrice];
+        accumulateBucketInterest(toBucket, _inflator);
+        toLpTokens = Maths.wdiv(_amount, getExchangeRate(toBucket));
+        toBucket.lpOutstanding += toLpTokens;
+        toBucket.onDeposit += removeFromDeposit;
+
+        if (_fromPrice > _toPrice) {
```
Comment (EdNoepel) [inline src/libraries/Buckets.sol:104]: This is going to cause interest accumulation regardless of whether the bucket is utilized.  We shouldn't need to reallocate anything if neither bucket is utilized.

**Review by EdNoepel — CHANGES_REQUESTED** [review]
The reallocation logic doesn't seem right.  When moving the price down, we must allocate that debt to the HUP downward.

Here are some truth tables to foster discussion:
| moving down		| from bucket utilized	   | from bucket unutilized |
| :--------------------| :-----------------------| :----------------------|
| to bucket utilized	| reallocateUp(to,from)   | not possible           |
| to bucket unutilized	| reallocateDown(from,lup)| no reallocation        |

| moving up		| from bucket utilized	| from bucket unutilized |
| :--------------------| :--------------------| :----------------------|
| to bucket utilized	| reallocateUp(from,to)| reallocateUp(lup,to)   |
| to bucket unutilized	| not possible		| no reallocation        |

I recommend we implement this in the simulator, such that we can examine the state of the book while interacting with it, to ensure we've gotten the reallocation logic right.

**Comment by grandizzy** [comment]
will be addressed in PR #90

**Pull request closed without merging**